packages/react-sdk/src/hook.ts

```diff
@@ -0,0 +1,42 @@
+import type { Instance, Prop } from "@webstudio-is/project-build";
+
+/**
+ * Hooks are subscriptions to builder events
+ * with limited way to interact with it.
+ * Called independently from components.
+ */
+
+export type HookContext = {
+  setPropVariable: (
+    instanceId: Instance["id"],
+    propName: Prop["name"],
+    value: unknown
+  ) => void;
+};
+
+export type InstanceSelection = Instance[];
+
+type NavigatorEvent = {
+  instanceSelection: InstanceSelection;
+};
+
+export type Hook = {
+  onNavigatorSelect?: (context: HookContext, event: NavigatorEvent) => void;
+  onNavigatorUnselect?: (context: HookContext, event: NavigatorEvent) => void;
+};
+
+export const getClosestInstance = (
```

Comment:
1. utils like this should have a test, so changing it is obvious and tests get added right away
2. I am surprised we don't have this utility in our instance utils we use in the builder,  I think all functional utilities which have to do with the data structure should be at some point in one reusable package. 
Reason is this is a foundation for plugins to manipulate our data structure. We need to start thinking about these utils as a public api for the data

Reply:
It's specific to hooks api. Internally we do not always map selector to instances.